tests/component.test.js

```diff
@@ -0,0 +1,163 @@
+const ParchKey = require('../model/parch-key.js');
+const Component = require('../model/component.js');
+const Port = require('../model/port.js');
+const Coord = require('../model/coord.js');
+
+//Suppress console logs
+console.log = jest.fn();
+
+var l1 = 'layer-1-id';
+var l2 = 'layer-2-id';
+var p1 = new Port('port-1-label', l1, new Coord(0, 0));
+var p2 = new Port('port-2-label', l2, new Coord(0, 5));
+
+test('initialize Component: parameters', () => {
+    let cVal = new Component('name', 'id', [l1, l2], 10, 10, 'entity', [new Port(), new Port()]);
+
+    expect(cVal.name).toBe('name');
+    expect(cVal.id).toBe('id');
+    expect(cVal.layers).toEqual([l1, l2]);
+    expect(cVal.xSpan).toBe(10);
+    expect(cVal.ySpan).toBe(10);
```

Comment:
Choose two different values here (10 and 20 perhaps) to avoid the issue where you accidentally set both the x and y to the x input parameter (which would still allow this to pass even though its incorrect).

Reply:
It's been changed.